src/main/scala/com/amazon/deequ/analyzers/RatioOfSums.scala

```diff
@@ -0,0 +1,81 @@
+package com.amazon.deequ.analyzers
+
+import com.amazon.deequ.analyzers.Preconditions.{hasColumn, isNumeric}
+import com.amazon.deequ.metrics.Entity
+import org.apache.spark.sql.DeequFunctions.stateful_corr
+import org.apache.spark.sql.{Column, Row}
+import org.apache.spark.sql.types.DoubleType
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.types.StructType
+import Analyzers._
+
+import com.amazon.deequ.metrics.Entity
+import com.amazon.deequ.repository.AnalysisResultSerde
+
+case class RatioOfSumsState(
+    numerator: Double,
+    denominator: Double
+) extends DoubleValuedState[RatioOfSumsState] {
+
+  override def sum(other: RatioOfSumsState): RatioOfSumsState = {
+    val n1 = numerator
+    val n2 = other.numerator
+    val newN = n1 + n2
+    val t1 = denominator
+    val t2 = other.denominator
+    val newD = t1 + t2
+    
+    RatioOfSumsState(newN, newD)
```

Comment:
Maybe we should rewrite this function to the one below. Looks unnecessary verbose.

```
override def sum(other: RatioOfSumsState): RatioOfSumsState = {
    RatioOfSumsState(numerator + other.numerator, denominator + other.denominator)
}
```

Reply:
I agree, but I was copying how the Correlation analyzer was written. However, looking through some other analyzers, such as Sum, they are written more concisely as you suggest. I will make this change.